internal/k8s/reconciler/listener_test.go

```diff
@@ -40,268 +42,355 @@ func TestListenerValidate(t *testing.T) {
 	defer ctrl.Finish()
 	client := mocks.NewMockClient(ctrl)
 
-	// protocols
-	listener := NewK8sListener(&gw.Gateway{}, gw.Listener{}, K8sListenerConfig{
-		Logger: hclog.NewNullLogger(),
+	t.Run("Unsupported protocol", func(t *testing.T) {
```

Comment:
The changes in this file are an attempt at breaking up the test into subtests so that it's easier to understand what's happening where. The only new test logic is in the subtests whose name mentions "cross-namespace" secret refs.